yaptide/application.py

```diff
@@ -50,6 +54,13 @@ def create_app():
 
     return app
 
+def check_submodules():
+    with open('.gitmodules') as f:
+        for line in f:
+            if 'path' in line:
```

Comment:
This is bad idea, it will find `path` in all comment lines and other unrelated line.

Reply:
You can check also idea suggested by chatgpt. What do you think are the pros and cons of each solution ?

```python
import subprocess

def check_submodules(repo_path):
    # Change the current working directory to the repo path
    subprocess.run(["cd", repo_path], check=True, shell=True)
    
    # Run the git submodule status command
    result = subprocess.run(["git", "submodule", "status"], capture_output=True, text=True, check=True)
    
    # Split the output into lines
    lines = result.stdout.strip().split('\n')
    
    # Check each line for uncloned or not updated submodules
    uncloned_submodules = [line for line in lines if line.startswith('-') or line.startswith('+')]
    
    if uncloned_submodules:
        print("There are submodules that haven't been cloned or are not up to date:")
        for submodule in uncloned_submodules:
            print(submodule)
    else:
        print("All submodules are cloned and up to date.")

# Replace '/path/to/repo' with the actual path to your repository
check_submodules('/path/to/repo')
```